### ISSUE TITLE
SSLv3, TLS 1, TLS 1.1, TLS1.2, and TLS 1.3 return Fixme: unexpected value

### ISSUE THREAD

**Issue by ivyheasty** [drwetter/testssl.sh#1134]
1. testssl version from the banner (testssl.sh -b 2>/dev/null | head -4 | tail -2)
testssl.sh 3.0rc1 from https://testssl.sh/dev/
(441f14a 2018-09-24 10:30:30 -- )

2. what exactly was happening, output is needed

The following is the output when executing this syntax : `./testssl.sh --jsonfile-pretty testssl_result.json {hostname}`

Output:

>  SSLv3   Fixme: unexpected value around line 4937, rerun w DEBUG>=2 or --ssl-native
>  TLS 1      Fixme: unexpected value around line 5005, rerun w DEBUG>=2 or --ssl-native
>  TLS 1.1    Fixme: unexpected value around line 5076, rerun w DEBUG>=2 or --ssl-native
>  TLS 1.2    Fixme: unexpected value around line 5158, rerun w DEBUG>=2 or --ssl-native
>  TLS 1.3    Fixme: unexpected value around line 5326, rerun w DEBUG>=2 or --ssl-native

3. what did you expect instead?
The above protocols are tested

4. steps to reproduce

   1. testssl.sh command line
`./testssl.sh --jsonfile-pretty testssl_result.json {hostname}`
 
    1. openssl version used (testssl.sh -b 2>/dev/null | head -16 | tail -3)
` Using "OpenSSL 1.0.2-chacha (1.0.2i-dev)" [~183 ciphers]
./bin/openssl.Darwin.x86_64
 (built: "Sep  7 19:34:54 2016", platform: "darwin64-x86_64-cc")`
 
    1. your operating system (uname -a)
`17.7.0 Darwin Kernel Version 17.7.0: Thu Jun 21 22:53:14 PDT 2018; root:xnu-4570.71.2~1/RELEASE_X86_64 x86_64`

**Comment by dcooper16** [comment]
I was able to reproduce these results. When I tested against testssl.net, I got the same results as above. When testing against google.com, there were no error messages, but all protocols were marked as "not offered".

It seems that the problem is https://github.com/drwetter/testssl.sh/commit/dd3657eb23c8234147cb9c8be8059213f856be9e. Using the new PRINTF works great on Linux, but it seems that it doesn't work on OS X. When testing on OS X I was able to fix the problem by changing `choose_printf()` to just set `PRINTF` to `printf`.

I believe this is also the reason for #1135.

**Comment by drwetter** [comment]
Couldn't reproduce this under FreeBSD. The major difference is the bash version though which doesn't make any sense to me.

 I'll check later.

**Comment by drwetter** [comment]
![screenshot_20180925_155609](https://user-images.githubusercontent.com/8036727/46019094-9bdb2000-c0db-11e8-9307-fc2a49731377.png)

**Comment by dcooper16** [comment]
I think I found out what is happening. I modified the code at the end of `socksend_tls_clienthello()` as follows by adding two lines in order to find out what was actually being printed to the socket when running the code on OS X:
```
     $PRINTF -- "$data" >&5 2>/dev/null &
     $PRINTF -- "$data" > out.bin 2>/dev/null
     printf -- "$data" > out2.bin 2>/dev/null
```
The third line produced what was expected:

![printf](https://user-images.githubusercontent.com/16493199/46026705-08ddbe00-c0ba-11e8-8610-add900c89ccf.png)

However, for the second line, when using `$PRINTF` instead of `printf`, I got the results below. It seems that "\\x" is just being interpreted as "x" rather than as an indication that the following two bytes should be processed as an ASCII-HEX encoding of a character. I believe it is the string below that is being sent to the server when using OS X rather than the one above.

![printf](https://user-images.githubusercontent.com/16493199/46026768-30348b00-c0ba-11e8-8e1b-de61e51428db.png)

**Comment by drwetter** [comment]
yeah, I came to the same conclusion. BSD's ``/usr/bin/printf`` is just different:

```
/usr/bin/printf --  "\x16\x03\x01\x02\x00\x01\x00\x01\xfc\x03" | xxd
00000000: 7831 3678 3033 7830 3178 3032 7830 3078  x16x03x01x02x00x
00000010: 3031 7830 3078 3031 7866 6378 3033       01x00x01xfcx03
```

Linux:
```
/usr/bin/printf -- "\x16\x03\x01\x02\x00\x01\x00\x1\xfc\x03" | xxd
0000000: 1603 0102 0001 0001 fc03                 ..........
```

Any BSD guy who understands the

**Comment by ivyheasty** [comment]
The issue is no longer reproducible using the latest commit. Thank you :)

**Comment by drwetter** [comment]
Closed related #1132 + #1135. Leave this open for now.

**Comment by drwetter** [comment]
see 1b52834dfcbc7da77fa69e0eca4df44be8cfa295
